rag/easy-rag/runtime/src/main/java/io/quarkiverse/langchain4j/easyrag/runtime/EasyRagRecorder.java

```diff
@@ -35,37 +38,77 @@ public void ingest(EasyRagConfig config, BeanContainer beanContainer) {
             LOGGER.info("Skipping document ingestion as per configuration");
             return;
         }
+
         EmbeddingStore<TextSegment> embeddingStore = beanContainer.beanInstance(EmbeddingStore.class);
         EmbeddingModel embeddingModel = beanContainer.beanInstance(EmbeddingModel.class);
 
+        if (config.reuseEmbeddings().enabled() && (embeddingStore instanceof InMemoryEmbeddingStore<TextSegment>)) {
+            Path embeddingsFile = Path.of(config.reuseEmbeddings().file()).toAbsolutePath();
+
+            // If the embeddings file already exists it would have been ingested
+            // when the InMemoryEmbeddingStore bean was created
+            // See the inMemoryEmbeddingStoreSupplier method
```

Comment:
One minor problem is that if you don't call anything on the `embeddingStore` object here, it won't get initialized, and thus the ingestion (from the file) will run lazily (when the store is actually accessed for the first time).  Could we add at least a dummy `embeddingStore.toString()` call or something like this to make sure that the file gets ingested eagerly at startup, for consistency?

Reply:
Couldn't you mark the `EmbeddingStore` bean as `@Startup` so that it gets initialized eagerly?

Reply:
Maybe changing its type from `ApplicationScoped` to `Singleton` (in the `EasyRagProcessor` class) would help? Not sure.
But no we can't add annotations directly on it, it's not from this codebase

Reply:
Maybe @geoand knows a better trick, but this is how I've been doing it :)

Reply:
That's how I do it as well

Reply:
Ok I've updated it and added some comments in the code.